### PR TITLE
Support type conversion for UDT fields (CSHARP-381)

### PR DESCRIPTION
**Pull request by landim**

### CODE REVIEW

File: src/Cassandra.IntegrationTests/Linq/LinqTable/LinqUdtTests.cs [hunk 1]
```diff
@@ -33,70 +29,112 @@ public override void OneTimeSetUp()
...
         }
 
         [Test, TestCassandraVersion(2, 1, 0)]
         public void LinqUdt_Select()
         {
             // Avoid interfering with other tests
-            _session.Execute("DROP TABLE IF EXISTS albums");
-            _session.Execute("CREATE TABLE albums (id uuid primary key, name text, songs list<frozen<song>>, publishingdate timestamp)");
+            _session.Execute("DROP TABLE IF EXISTS band");
+            _session.Execute("CREATE TABLE band (id uuid primary key, name text, albuns list<frozen<album>>)");
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Linq/LinqTable/LinqUdtTests.cs]: typo "albuns"

File: src/Cassandra.IntegrationTests/Linq/LinqTable/LinqUdtTests.cs [hunk 2]
```diff
@@ -33,70 +29,112 @@ public override void OneTimeSetUp()
...
             Assert.NotNull(songs);
             Assert.AreEqual(2, songs.Count);
-            Assert.NotNull(songs.FirstOrDefault(s => s.Title == "Good Times Bad Times"));
-            Assert.NotNull(songs.FirstOrDefault(s => s.Title == "Communication Breakdown"));
+            Assert.NotNull(songs.FirstOrDefault(s => s.Title == "Monkey Wrench"));
+            Assert.NotNull(songs.FirstOrDefault(s => s.Title == "Everlong"));
         }
+
+//        [Test, TestCassandraVersion(2, 1, 0)]
+//        public void LinqUdt_Select_Enum()
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Linq/LinqTable/LinqUdtTests.cs]: Left commented block

File: src/Cassandra.IntegrationTests/Linq/Structures/Band.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿// //
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Linq/Structures/Band.cs]: `//` appears twice (we can fix the resharper macro later, but the source code should be clean)

File: src/Cassandra.IntegrationTests/Linq/Structures/Band.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using Cassandra.Tests.Mapping.Pocos;
+
+namespace Cassandra.IntegrationTests.Linq.Structures
+{
+    public class Band
+    {
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public List<Album> Albuns { get; set; }
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Linq/Structures/Band.cs]: Same typo

**Comment by landim** [comment]
Addressed review issues:
- Removed unnecessary test
- Removed block of commented code
- Fix code minor warnings

**Comment by jorgebay** [comment]
Test changes looks good to me, I'll merge it!

Thanks @ElliotSis !

**Pull request merged**